src/bitcoin/poller/looper.rs

```diff
@@ -354,6 +354,45 @@ pub fn maybe_initialize_tip(bit: &impl BitcoinInterface, db: &impl DatabaseInter
     }
 }
 
+/// Initialize the wallet with DB data, if required.
+pub fn maybe_initialize_wallet(bit: &mut impl BitcoinInterface, db: &impl DatabaseInterface) {
```

Comment:
Why doing this in the poller rather than at startup with the rest of the initialization?

Reply:
That's true, I was originally loading DB data as part of `setup_electrum` and didn't need this step in the poller. But after the saving txs work, I was thinking we might one day need to read data from Electrum for a DB migration before calling `setup_sqlite` and so we might need to call `setup_electrum` before the DB is available.

In that case, however, we'd only need the Electrum client for reading data from the Electrum server and not the `BitcoinInterface` necessarily, similarly to how the migration currently creates a `BitcoinD` object, so perhaps that would still be fine.

If you agree, I'll just do this initialization as part of `setup_electrum` and remove these parts from the poller and `BitcoinInterface`.

Reply:
As mentioned in https://github.com/wizardsardine/liana/pull/1222#discussion_r1711522779, I've now removed the initialization logic from the poller and `BitcoinInterface`.

Also, the `setup_electrum` function now contructs the Electrum client and `BdkWallet` separately as this seemed more natural (and avoided many arguments). So it'll be easier to use the Electrum client for any future DB migration before creating the wallet.